RubberduckTests/CodeExplorer/CodeExplorerComparerTests.cs

```diff
@@ -0,0 +1,357 @@
+            var folderNode2 = new CodeExplorerCustomFolderViewModel(null, "Name2", "Name2", null, Enumerable.Empty<Declaration>());
+
+            Assert.IsTrue(new CompareByName().Compare(folderNode1, folderNode2) < 0);
+        }
+
+        [Category("Code Explorer")]
+        [Test]
+        public void CompareByType_ReturnsZeroForIdenticalNodes()
+        {
+            var errorNode = new CodeExplorerCustomFolderViewModel(null, "Name", "folder1.folder2", null, Enumerable.Empty<Declaration>());
+            Assert.AreEqual(0, new CompareByName().Compare(errorNode, errorNode));
+        }
+
+        [Category("Code Explorer")]
+        [Test]
+        public void CompareByType_ReturnsEventAboveConst()
+        {
+            const string inputCode =
+@"Public Event Foo(ByVal arg1 As Integer, ByVal arg2 As String)
+Public Const Bar = 0";
+
+            using (var explorer = new MockedCodeExplorer(inputCode)
+                .SelectFirstModule())
+            {
+                var module = explorer.ViewModel.SelectedItem;
+                var eventNode = module.Children.Single(s => s.Name == "Foo");
+                var constNode = module.Children.Single(s => s.Name == "Bar");
+
+                Assert.AreEqual(-1, new CompareByDeclarationType().Compare(eventNode, constNode));
+            }
+        }
+
+        [Category("Code Explorer")]
+        [Test]
+        public void CompareByType_ReturnsConstAboveField()
+        {
+            const string inputCode =
+@"Public Const Foo = 0
+Public Bar As Boolean";
+
+            using (var explorer = new MockedCodeExplorer(inputCode)
+                .SelectFirstModule())
+            {
+                var module = explorer.ViewModel.SelectedItem;
+                var constNode = module.Children.Single(s => s.Name == "Foo");
+                var fieldNode = module.Children.Single(s => s.Name == "Bar");
+
+                Assert.AreEqual(-1, new CompareByDeclarationType().Compare(constNode, fieldNode));
+            }
+        }
+
+        [Category("Code Explorer")]
+        [Test]
+        public void CompareByType_ReturnsFieldAbovePropertyGet()
+        {
+            const string inputCode =
+@"Private Bar As Boolean
+
+Public Property Get Foo() As Variant
+End Property
+";
+
+            using (var explorer = new MockedCodeExplorer(inputCode)
+                .SelectFirstModule())
+            {
+                var module = explorer.ViewModel.SelectedItem;
+                var fieldNode = module.Children.Single(s => s.Name == "Bar");
+                var propertyGetNode = module.Children.Single(s => s.Name == "Foo (Get)");
+
+                Assert.AreEqual(-1, new CompareByDeclarationType().Compare(fieldNode, propertyGetNode));
+            }
+        }
+
+        [Category("Code Explorer")]
+        [Test]
+        public void CompareByType_ReturnsPropertyGetEqualToPropertyLet()
```

Comment:
This set of tests could have been done as a data-driven tests. Something like:

```[TestCase(0, "Foo (Get)", "Foo (Let)")]
[TestCase, 0, "Foo (Get)", "Foo (Set)")]
...
public void CompareByType_ReturnsPropertyEqual(int expected, string name1, string name2)
```

That would significantly cut down on the amount of copy'n'pasting and more importantly make the tests more maintainable.

Reply:
Agreed. The old UTs could probably still use some more clean-up. This is low priority though, so I'll flag it with a TODO in case I forget to circle back to it.